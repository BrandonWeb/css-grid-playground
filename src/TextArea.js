import React from 'react';
import styled from 'styled-components';

const TextAreaField = styled.textarea`
  border: 0px solid black;
  padding: 0;
  outline: none;
  background: none;
  resize: none;
  height: 100%;
  width: 100%;
  font-family: Monaco, monospace;
  @media (max-width: 600px) {
    font-size: .75rem;
  }
  ${props => props.styling}
`;

class TextArea extends React.Component {
  static propTypes = {
    styling: React.PropTypes.string,
    value: React.PropTypes.string,
    onChange: React.PropTypes.func,
  };
  static defaultProps = {
    styling: '',
    value: '',
    onChange: null,
  }
  handleChange = e => this.props.onChange(e.target.value)
  render() {
    return (
      <TextAreaField
        styling={this.props.styling}
        value={this.props.value}
        onChange={this.handleChange}
      />
    );
  }
}

export default TextArea;
