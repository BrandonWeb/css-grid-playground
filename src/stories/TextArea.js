import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import TextArea from '../TextArea';

const stories = storiesOf('TextArea', module);

stories.add('plain textarea', () => <TextArea />);
stories.add('with styling', () => <TextArea styling={`background: red;`} />);
stories.add('with value', () => <TextArea value={`display: grid;`} />);
stories.add('with onChange', () => <TextArea onChange={action('onChange')} />);
