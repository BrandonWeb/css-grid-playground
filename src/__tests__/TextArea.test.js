import React from 'react';
import renderer from 'react-test-renderer';
import TextArea from '../TextArea';

it('renders correctly', () => {
  const value = `background: red;
width: 100px;`;

  const Component = (
    <TextArea
      value={value}
      onChange={() => {}}
    />
  );
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
});
