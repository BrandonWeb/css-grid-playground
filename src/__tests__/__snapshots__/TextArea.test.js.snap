exports[`test renders correctly 1`] = `
<textarea
  className="kEAiXh"
  defaultValue=""
  onChange={[Function]} />
`;
